### ISSUE TITLE
TypeError: this.util.TextEncoder is not a constructor

### ISSUE THREAD

**Issue by shujip** [hiukim/mind-ar-js-react#4]
when i import these in my react project as example:

import * as three from "three";
import 'mind-ar/dist/mindar-image-three.prod.js';
import 'mind-ar/dist/mindar-image.prod.js';
import 'aframe';
import 'mind-ar/dist/mindar-image-aframe.prod.js';

i get error:

TypeError: this.util.TextEncoder is not a constructor

what should I do?

**Comment by nickgrealy** [comment]
I had the same problem, albeit with Gatsby + ReactJS + Node16.

The problem was `mind-ar/dist/mindar-image-three.prod.js` is a CLIENT-ONLY script. When Gatsby was compiling and trying to resolve dependencies at build time, I was getting this error. 

I tried mucking around with webpack - I suspected it needed a 'utils' polyfill (the code references `this.util = require('util')` - my recommendation, don't waste your time.

**TLDR;** My solution, was to programatically import it, IF the client was on the browser (only).

Additionally, it couldn't be imported as a module (e.g. `import MINDAR from '...'`) - so I had to wait until it was loaded.

Here's my solution, with addtional Typescript bindings for "window".

```ts
// import CLIENT-ONLY library
if (typeof window !== 'undefined') import("mind-ar/dist/mindar-image-three.prod.js" as any)

declare global {
  interface Window { MINDAR: any; }
}

const getMindAR = async () => waitFor(() => {
  console.log("waiting for mindar...", {  mindar: window.MINDAR })
  return  window.MINDAR
})
```

Note: `waitFor` was a polling promise library, that waited until a condition was met.
